From 63e5c19737fcf33198077c2a5c4ca65d478b1982 Mon Sep 17 00:00:00 2001
From: Guilhem Moulin <guilhem@debian.org>
Date: Mon, 23 Dec 2024 17:01:31 +0100
Subject: [PATCH] debdiff: Always call dpkg-source(1) when `--apply-patches` is
 set.

The .orig tarballs need to be unpacked in patch-applied mode, regardless
of the DSC format or whether the .orig tarballs are identical.

Regression from c583b086f9d83c35556aee8c9d84bff6385d66e9.

Closes: #-1
---
 scripts/debdiff.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/debdiff.pl b/scripts/debdiff.pl
index e12a0d3e..878cf354 100755
--- a/scripts/debdiff.pl
+++ b/scripts/debdiff.pl
@@ -720,7 +720,7 @@ if ($type eq 'deb') {
             push @opts, '--skip-patches';
         }
         my $diri = ${"dir$i"};
-        if ($origs[1] eq $origs[2] and $dscformats[$i] eq '3.0 (quilt)') {
+        if ($origs[1] eq $origs[2] and $dscformats[$i] eq '3.0 (quilt)' and !$apply_patches) {
             eval {
                 my $source = $origs[$i];
                 $source =~ s/\.orig\.tar\.$compression_re//;
-- 
2.45.2

